apps/api/app/workers/tasks/tracked_todo_tasks.py

```
@@ -0,0 +1,416 @@
+                    ref_canvas = await MongoVFS().read(
+                        path=f"{ref_doc['vfs_path']}/canvas.md", user_id=user_id
+                    )
+                    if ref_canvas and "## Learnings" in ref_canvas:
+                        learnings_start = ref_canvas.index("## Learnings")
+                        next_section = ref_canvas.find("\n## ", learnings_start + 1)
+                        learnings = ref_canvas[learnings_start:next_section] if next_section != -1 else ref_canvas[learnings_start:]
+                        ref_parts.append(
+                            f"From past todo \"{ref_doc.get('title', 'Unknown')}\":\n{learnings.strip()}"
+                        )
+            except Exception:
+                continue
```

Ruff (0.15.6)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)